frontends/verilog/verilog_parser.y

```diff
@@ -2522,6 +2522,7 @@ gen_stmt:
 	} simple_behavioral_stmt ';' expr {
 		ast_stack.back()->children.push_back($6);
 	} ';' simple_behavioral_stmt ')' gen_stmt_block {
+		SET_AST_NODE_LOC(ast_stack.back(), @1, @9);
```

Comment:
If I'm counting correctly, the location of the `AST_GENFOR` node should begin with the location of the `TOK_FOR` (OK, I agree) and end with the location of the final `simple_behavioral_stmt`, i.e. the `i++` in `for(i = 0; i < 10; i++)`?

Should it not include the contents of the `gen_stmt_block` at `@11`?  I am honestly not sure.

Reply:
Ahh, does the counting include actions as well as tokens?

Reply:
Yep.  See: https://www.gnu.org/software/bison/manual/html_node/Using-Midrule-Actions.html

```
The midrule action itself counts as one of the components of the rule. This makes a difference when there is another action later in the same rule (and usually there is another at the end): you have to count the actions along with the symbols when working out which number n to use in $n. 
```

Reply:
Thanks for clearing up my misunderstanding! I've recounted them all and I think all the review comments should be cleared up, please could you take another look?